client/admin/customEmoji/AddCustomEmoji.js

```diff
@@ -0,0 +1,141 @@
+import React, { useState } from 'react';
+import { Box, Button, ButtonGroup, Margins, TextInput, Field, InputBox } from '@rocket.chat/fuselage';
+import { useUniqueId } from '@rocket.chat/fuselage-hooks';
+import s from 'underscore.string';
+
+import { useTranslation } from '../../contexts/TranslationContext';
+import { useMethod } from '../../contexts/ServerContext';
+import { useToastMessageDispatch } from '../../contexts/ToastMessagesContext';
+
+export function AddCustomEmojis({ goToNew, close, onChange, ...props }) {
+	const t = useTranslation();
+	const dispatchToastMessage = useToastMessageDispatch();
+
+	// const [name, setName] = useState('');
+	const [aliases, setAliases] = useState('');
+
+	const uploadCustomEmoji = useMethod('uploadCustomEmoji');
+
+	const saveEmoji = useMethod('insertOrUpdateUserStatus');
+
+	const [newData, setNewData] = useState({});
+
+	const fileSourceInputId = useUniqueId();
+
+	const createEmojiData = (name) => {
+		const soundEmoji = {};
+		soundEmoji.name = s.trim(name);
+		soundEmoji.newFile = true;
+		return soundEmoji;
+	};
+
+	const saveAction = async (newData) => {
+		const soundData = createEmojiData(newData.name);
+		let soundId;
+
+		try {
+			soundId = await saveEmoji(soundData);
```

Comment:
I don't believe that this method returns an id, this was a thing very specific to the custom sound methods, in here you will have to check what does the method `uploadEmojiCustom` do and what it needs. From what I've read it needs something like `reader.result, newData.soundFile.type, soundData`